### PR TITLE
[CEN-1282] Add cosmos db (mongo) with a transaction database

### PR DESCRIPTION
**Pull request by GiovanniMancini**
This PR proposes to add a Cosmos/Mongo account + a transaction DB to support ingestion pipeline of TAE application

### List of changes

- New Cosmos/Mongo account
- New Transaction DB
- New dedicated subnet
- New private dns zone to resolve private endpoint

### Motivation and context

This PR is required to have a scratch db where transactions can be ingested before aggregation for ADE

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/database.tf [hunk 1]
```diff
@@ -128,7 +128,7 @@ resource "azurerm_cosmosdb_mongo_database" "transaction" {
 
   name                = "transaction"
   resource_group_name = azurerm_resource_group.db_rg.name
-  account_name        = module.cosmosdb_account_mongodb[0].name
+  account_name        = module.cosmosdb_account_mongodb[count.index].name
```
Comment (uolter) [inline src/database.tf:131]: it's ok, but since count can only assumes 0 or 1 .... it does not change that much.
It's a style exercise ;-)

**Review by uolter — DISMISSED** [review]

File: src/database.tf (hunk 1 above)
Comment (uolter) [reply src/database.tf:131]: @GiovanniMancini  mmmm well I changed my mind...
When count = 0 the resource **azurerm_cosmosdb_mongo_database.transaction** does not exist
while when count = 1 the resource exists but I feel module.cosmosdb_account_mongodb[1].name is throwing an error: the index 1 does not exist.
Am I wrong?

File: src/env/prod/terraform.tfvars [hunk 2]
```diff
@@ -292,6 +293,44 @@ db_metric_alerts = {
...
+  #   failover_priority = 1
+  #   zone_redundant    = false
+  # }]
+
+  additional_geo_locations          = []
+  private_endpoint_enabled          = true
+  public_network_access_enabled     = false
+  is_virtual_network_filter_enabled = true
+
+  backup_continuous_enabled = false
```
Comment (pasqualedevita) [inline src/env/prod/terraform.tfvars:323]: for production envs use continuous backup

File: src/env/prod/terraform.tfvars [hunk 3]
```diff
@@ -292,6 +293,44 @@ db_metric_alerts = {
     dimension   = []
   }
 }
+
+cosmos_mongo_db_params = {
+  enabled      = false
+  kind         = "MongoDB"
+  capabilities = ["EnableMongo", "EnableServerless"]
```
Comment (pasqualedevita) [inline src/env/prod/terraform.tfvars:300]:
```suggestion
  capabilities = ["EnableMongo"]
```
serverless is for dev/test envs. For production with heavy loads better use provisioned throughput

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Review by diegolagospagopa — DISMISSED** [review]

**Pull request merged**